### ISSUE TITLE
The code appears to be wrong? 

### ISSUE THREAD

**Issue by dabeeeenster** [pulumi/docs#3617]
File: [docs/intro/concepts/config.md](https://www.pulumi.com/docs/intro/concepts/config/)

In python, config.require() gets my secret config, but config.require_secret fails with `TypeError: Object of type Output is not JSON serializable` - just require() appears to work too which is odd? That's not what these docs say?

**Comment by EvanBoyle** [comment]
Can you share the full code snippet that's causing the error? `require()` will return a plain type like `string` where `require_secret` will return that same type wrapped in an `Output`. Outputs contain information about secretness that the pulumi engine can propagate across the RPC boundary. If you try to serialize an output (json.dumps) you might see an error like this.

**Comment by dabeeeenster** [comment]
Ah understood, thanks. The docs don't make that clear at all IMO:

```
print(config.require('name'))
print(config.require_secret('dbPassword'))
```

In this example, we have read back the name and dbPassword configuration variables programmatically. The name is just the string BroomeLLC, while the dbPassword is a secret output value that is encrypted.

Sounds wrong?

**Comment by EvanBoyle** [comment]
Yeah, any ouputs need to be printed within an Apply() call. I've updated the title to reflect the issue.
